simpleclient_spring_boot/src/main/java/io/prometheus/client/spring/boot/PrometheusMVCEndpoint.java

```diff
@@ -0,0 +1,33 @@
+package io.prometheus.client.spring.boot;
+
+import org.springframework.boot.actuate.endpoint.mvc.EndpointMvcAdapter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@ConfigurationProperties("endpoints.prometheus")
+public class PrometheusMVCEndpoint extends EndpointMvcAdapter {
+
+	private final PrometheusEndpoint delegate;
+	
+	public PrometheusMVCEndpoint(PrometheusEndpoint delegate) {
+		super(delegate);
+		this.delegate = delegate;
+	}
+	
+	@ResponseBody
+	@RequestMapping(value="/metrics", produces={"text/plain;version=0.0.4"})
+	public String metrics() {
+		if (!this.delegate.isEnabled()) {
+			// Shouldn't happen - MVC endpoint shouldn't be registered when delegate's
+			// disabled
+			return getDisabledResponse().toString();
```

Comment:
Change the return value to `Object` and remove the `toString` (otherwise you lose the 404 status code)